crates/libs/sys/src/lib.rs

```diff
@@ -7,7 +7,9 @@ Learn more about Rust for Windows here: <https://github.com/microsoft/windows-rs
 #![allow(non_snake_case, non_upper_case_globals, non_camel_case_types, clippy::all)]
 #![cfg_attr(not(feature = "docs"), doc(hidden))]
 
+#[allow(unused_extern_crates)]
 extern crate self as windows_sys;
```

Comment:
It does not seem that this is needed for anything; the lint is valid. There is nothing inside `sys` / `windows` referring to `(::)windows_sys::` or `(::)windows::` respectively. Could these lines just be removed altogether?

Reply:
Err, I meant to say that:

1. On `windows_sys` I don't seem to get a false-positive here, because it's actually used;
2. On `windows` I only get a few errors originating from the `windows_implement` macro, which expects the `windows` crate to be available as an external crate (which is interesting for a macro that generates code to be used both inside and outside of a crate).

Reply:
```
> cargo clippy -p windows-sys
    Checking windows-sys v0.52.0 (R:\windows-rs\crates\libs\sys)
warning: unused extern crate
  --> crates\libs\sys\src\lib.rs:11:1
   |
11 | extern crate self as windows_sys;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: remove it
   |
   = note: `-W unused-extern-crates` implied by `-W rust-2018-idioms`
   = help: to override `-W rust-2018-idioms` add `#[allow(unused_extern_crates)]`

warning: `windows-sys` (lib) generated 1 warning (run `cargo clippy --fix --lib -p windows-sys` to apply 1 suggestion)
    Finished dev [unoptimized + debuginfo] target(s) in 2m 49s
```

Reply:
See the `console` highlight:

> ```console
> > cargo clippy -p windows-sys
>     Checking windows-sys v0.52.0 (R:\windows-rs\crates\libs\sys)
> warning: unused extern crate
>   --> crates\libs\sys\src\lib.rs:11:1
>    |
> 11 | extern crate self as windows_sys;
>    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: remove it
>    |
>    = note: `-W unused-extern-crates` implied by `-W rust-2018-idioms`
>    = help: to override `-W rust-2018-idioms` add `#[allow(unused_extern_crates)]`
> 
> warning: `windows-sys` (lib) generated 1 warning (run `cargo clippy --fix --lib -p windows-sys` to apply 1 suggestion)
>     Finished dev [unoptimized + debuginfo] target(s) in 2m 49s
> 
> > cargo clippy -p windows-sys --features Win32
>     Checking windows-sys v0.52.0 (R:\windows-rs\crates\libs\sys)
>     Finished dev [unoptimized + debuginfo] target(s) in 1.41s
> ```

Indeed, it is conditionally used :(